app/routes/__index/certificate/index.tsx

```diff
@@ -1,43 +1,75 @@
 import { Flex, Center } from '@chakra-ui/react';
-import type { Certificate } from '@prisma/client';
-import { useState } from 'react';
+import type { LoaderArgs, ActionArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
+import { typedjson, useTypedLoaderData } from 'remix-typedjson';
+import { useRevalidator } from '@remix-run/react';
+import { useInterval } from 'react-use';
 
-import undrawSvg from '~/assets/undraw_processing_re_tbdu.svg';
+import { requireUser } from '~/session.server';
+import pendingSvg from '~/assets/undraw_processing_re_tbdu.svg';
 import Loading from '~/components/display-page';
 import CertificateAvailable from '~/components/certificate/certificate-available';
 import CertificateRequestView from '~/components/certificate/certificate-request';
 import { useUser } from '~/utils';
+import { getCertificateStatusByUsername } from '~/models/certificate.server';
+import { addCertRequest } from '~/queues/certificate/certificate-flow.server';
 
-export default function CertificateIndexRoute() {
-  const user = useUser();
-  const [certificateRequested, setCertificateRequested] = useState(false);
-  const [loading, setLoading] = useState(false);
+export const loader = async ({ request }: LoaderArgs) => {
+  const user = await requireUser(request);
+  let certificate;
+
+  try {
+    certificate = await getCertificateStatusByUsername(user.username);
+  } catch {
+    certificate = {
```

Comment:
I would just let certificate stay undefined.  Below, you can use `certificate?.status` to check if there is an Object before you try to access `.status`